source/_UIAHandler.py

```diff
@@ -224,9 +230,40 @@ def MTAThreadFunc(self):
 			self.MTAThreadInitException=e
 		finally:
 			self.MTAThreadInitEvent.set()
-		self.MTAThreadStopEvent.wait()
+		while True:
+			func=self.MTAThreadQueue.get()
+			if func:
+				func()

```

Comment:
Could a try/except handler help here? While you use exception handling in the only func that is currently queued, an unexpected failure can just bring down the whole thread.